assets/js/modules/analytics-4/datastore/properties.js

```diff
@@ -490,6 +501,58 @@ const baseActions = {
 			.setGoogleTagContainerID( googleTagContainerID );
 		registry.dispatch( MODULES_ANALYTICS_4 ).setGoogleTagID( googleTagID );
 	},
+
+	/**
+	 * Syncs Google Tag settings.
+	 *
+	 * @since n.e.x.t
+	 */
+	*syncGoogleTagSettings() {
+		if ( ! isFeatureEnabled( 'gteSupport' ) ) {
+			return;
+		}
+
+		const { select, dispatch, __experimentalResolveSelect } =
+			yield Data.commonActions.getRegistry();
+
+		// Wait for modules to be available before selecting.
+		yield Data.commonActions.await(
+			__experimentalResolveSelect( CORE_MODULES ).getModules()
+		);
+
+		const { isModuleConnected } = select( CORE_MODULES );
+
+		if ( ! isModuleConnected( 'analytics-4' ) ) {
+			return;
+		}
+
+		// Wait for module settings to be available before selecting.
+		yield Data.commonActions.await(
+			__experimentalResolveSelect( MODULES_ANALYTICS_4 ).getSettings()
+		);
+
+		const { getGoogleTagID, getMeasurementID, getGoogleTagLastSyncedAtMs } =
+			select( MODULES_ANALYTICS_4 );
+
+		const googleTagID = getGoogleTagID();
+		const measurementID = getMeasurementID();
+		const googleTagLastSyncedAtMs = getGoogleTagLastSyncedAtMs();
+
+		if (
+			isEmpty( googleTagID ) &&
```

Comment:
As mentioned in my reply to your question, I still don't see any reason to use `isEmpty()` here over a regular `! googleTagID`... Calling `isEmpty()` should have the same result but there's no need to use a third-party utility...